--- mate-power-manager-1.5.1/man/Makefile.am~	2012-11-23 04:26:10.000000000 +0200
+++ mate-power-manager-1.5.1/man/Makefile.am	2013-02-16 23:48:58.764862729 +0200
@@ -12,13 +12,13 @@
 
 if HAVE_DOCBOOK2MAN
 mate-power-manager.1: mate-power-manager.sgml
-	docbook2man $? &> /dev/null
+	docbook2man $?
 
 mate-power-preferences.1: mate-power-preferences.sgml
-	docbook2man $? &> /dev/null
+	docbook2man $?
 
 mate-power-statistics.1: mate-power-statistics.sgml
-	docbook2man $? &> /dev/null
+	docbook2man $?
 endif
 
 clean-local :
