diff -upr mate-power-manager-orig/src/gpm-control.c mate-power-manager-test/src/gpm-control.c
--- mate-power-manager-orig/src/gpm-control.c	2013-11-01 01:49:11.015502000 +0100
+++ mate-power-manager-test/src/gpm-control.c	2013-11-08 11:41:38.444731266 +0100
@@ -288,10 +288,11 @@ gpm_control_suspend (GpmControl *control
 		}
 		g_object_unref(proxy);
 	}
+#if !UP_CHECK_VERSION(0, 99, 0)
 	else {
 		ret = up_client_suspend_sync (control->priv->client, NULL, error);
 	}
-
+#endif
 	egg_debug ("emitting resume");
 	g_signal_emit (control, signals [RESUME], 0, GPM_CONTROL_ACTION_SUSPEND);
 
@@ -398,10 +399,11 @@ gpm_control_hibernate (GpmControl *contr
 			ret = TRUE;
 		}
 	}
+#if !UP_CHECK_VERSION(0, 99, 0)
 	else {
 		ret = up_client_hibernate_sync (control->priv->client, NULL, error);
 	}
-
+#endif
 	egg_debug ("emitting resume");
 	g_signal_emit (control, signals [RESUME], 0, GPM_CONTROL_ACTION_HIBERNATE);
 
diff -upr mate-power-manager-orig/src/gpm-engine.c mate-power-manager-test/src/gpm-engine.c
--- mate-power-manager-orig/src/gpm-engine.c	2013-10-14 22:33:51.547439000 +0200
+++ mate-power-manager-test/src/gpm-engine.c	2013-11-08 14:37:29.681600913 +0100
@@ -805,12 +805,14 @@ gpm_engine_coldplug_idle_cb (GpmEngine *
 	guint i;
 	GPtrArray *array = NULL;
 	UpDevice *device;
+#if !UP_CHECK_VERSION(0, 99, 0)
 	gboolean ret;
 	GError *error = NULL;
+#endif
 
 	g_return_val_if_fail (engine != NULL, FALSE);
 	g_return_val_if_fail (GPM_IS_ENGINE (engine), FALSE);
-
+#if !UP_CHECK_VERSION(0, 99, 0)
 	/* get devices from UPower */
 	ret = up_client_enumerate_devices_sync (engine->priv->client, NULL, &error);
 	if (!ret) {
@@ -818,7 +820,7 @@ gpm_engine_coldplug_idle_cb (GpmEngine *
 		g_error_free (error);
 		goto out;
 	}
-
+#endif
 	/* connected mobile phones */
 	gpm_phone_coldplug (engine->priv->phone);
 
@@ -831,7 +833,9 @@ gpm_engine_coldplug_idle_cb (GpmEngine *
 		gpm_engine_device_add (engine, device);
 		gpm_engine_check_recall (engine, device);
 	}
+#if !UP_CHECK_VERSION(0, 99, 0)
 out:
+#endif
 	if (array != NULL)
 		g_ptr_array_unref (array);
 	/* never repeat */
diff -upr mate-power-manager-orig/src/gpm-prefs-core.c mate-power-manager-test/src/gpm-prefs-core.c
--- mate-power-manager-orig/src/gpm-prefs-core.c	2013-11-01 01:49:00.864400000 +0100
+++ mate-power-manager-test/src/gpm-prefs-core.c	2013-11-08 15:40:30.067567455 +0100
@@ -810,7 +810,9 @@ gpm_prefs_init (GpmPrefs *prefs)
 	UpDevice *device;
 	UpDeviceKind kind;
 	GpmBrightness *brightness;
+#if !UP_CHECK_VERSION(0, 99, 0)
 	gboolean ret;
+#endif
 	guint i;
 
 	GDBusProxy *proxy;
@@ -896,9 +898,11 @@ gpm_prefs_init (GpmPrefs *prefs)
 	else {
 		/* are we allowed to shutdown? */
 		egg_console_kit_can_stop (prefs->priv->console, &prefs->priv->can_shutdown, NULL);
+#if !UP_CHECK_VERSION(0, 99, 0)
 		/* get values from UpClient */
 		prefs->priv->can_suspend = up_client_get_can_suspend (prefs->priv->client);
 		prefs->priv->can_hibernate = up_client_get_can_hibernate (prefs->priv->client);
+#endif
 	}
 
 	if (LOGIND_RUNNING()) {
@@ -952,14 +956,14 @@ gpm_prefs_init (GpmPrefs *prefs)
 	brightness = gpm_brightness_new ();
 	prefs->priv->has_lcd = gpm_brightness_has_hw (brightness);
 	g_object_unref (brightness);
-
+#if !UP_CHECK_VERSION(0, 99, 0)
 	/* get device list */
 	ret = up_client_enumerate_devices_sync (prefs->priv->client, NULL, &error);
 	if (!ret) {
 		egg_warning ("failed to get device list: %s", error->message);
 		g_error_free (error);
 	}
-
+#endif
 	devices = up_client_get_devices (prefs->priv->client);
 	for (i=0; i<devices->len; i++) {
 		device = g_ptr_array_index (devices, i);
diff -upr mate-power-manager-orig/src/gpm-statistics.c mate-power-manager-test/src/gpm-statistics.c
--- mate-power-manager-orig/src/gpm-statistics.c	2013-10-14 22:33:51.552439000 +0200
+++ mate-power-manager-test/src/gpm-statistics.c	2013-11-08 14:38:15.855069197 +0100
@@ -1794,11 +1794,12 @@ main (int argc, char *argv[])
 
 	wakeups = up_wakeups_new ();
 	g_signal_connect (wakeups, "data-changed", G_CALLBACK (gpm_stats_data_changed_cb), NULL);
-
+#if !UP_CHECK_VERSION(0, 99, 0)
 	/* coldplug */
 	ret = up_client_enumerate_devices_sync (client, NULL, NULL);
 	if (!ret)
 		goto out;
+#endif
 	devices = up_client_get_devices (client);
 
 	/* add devices in visually pleasing order */
@@ -1854,8 +1855,9 @@ main (int argc, char *argv[])
 	gtk_widget_show (widget);
 
 	gtk_main ();
-
+#if !UP_CHECK_VERSION(0, 99, 0)
 out:
+#endif
 	g_object_unref (settings);
 	g_object_unref (client);
 	g_object_unref (wakeups);
