diff -upr mate-power-manager-orig/src/gpm-engine.c mate-power-manager-test/src/gpm-engine.c
--- mate-power-manager-orig/src/gpm-engine.c	2013-11-09 21:19:14.952669000 +0100
+++ mate-power-manager-test/src/gpm-engine.c	2013-11-09 22:45:14.729415620 +0100
@@ -868,6 +868,21 @@ gpm_engine_device_added_cb (UpClient *cl
  * gpm_engine_device_removed_cb:
  **/
 static void
+#if UP_CHECK_VERSION(0, 99, 0)
+gpm_engine_device_removed_cb (UpClient *client, const char *object_path, GpmEngine *engine)
+{
+        guint i;
+
+        for (i = 0; i < engine->priv->array->len; i++) {
+                UpDevice *device = g_ptr_array_index (engine->priv->array, i);
+
+                if (g_strcmp0 (object_path, up_device_get_object_path (device)) == 0) {
+                        g_ptr_array_remove_index (engine->priv->array, i);
+                        break;
+                }
+        }
+}
+#else
 gpm_engine_device_removed_cb (UpClient *client, UpDevice *device, GpmEngine *engine)
 {
 	gboolean ret;
@@ -876,7 +891,7 @@ gpm_engine_device_removed_cb (UpClient *
 		return;
 	gpm_engine_recalculate_state (engine);
 }
-
+#endif
 
 /**
  * gpm_engine_device_changed_cb:
