diff -upr mate-power-manager-orig/src/gpm-manager.c mate-power-manager-test/src/gpm-manager.c
--- mate-power-manager-orig/src/gpm-manager.c	2013-11-09 21:19:14.954669000 +0100
+++ mate-power-manager-test/src/gpm-manager.c	2013-11-10 15:19:23.540948886 +0100
@@ -972,9 +972,9 @@ gpm_manager_get_spindown_timeout (GpmMan
  **/
 static void
 #if UP_CHECK_VERSION(0, 99, 0)
-gpm_manager_client_changed_cb (UpClient *client, GpmManager *manager)
-#else
 gpm_manager_client_changed_cb (UpClient *client, GParamSpec *pspec, GpmManager *manager)
+#else
+gpm_manager_client_changed_cb (UpClient *client, GpmManager *manager)
 #endif
 {
 	gboolean event_when_closed;
