--- mate-power-manager-1.5.1/applets/brightness/Makefile.am~	2013-03-02 19:33:45.000000000 +0200
+++ mate-power-manager-1.5.1/applets/brightness/Makefile.am	2013-03-02 19:35:13.858468368 +0200
@@ -60,7 +60,7 @@
             -e "s|\@LIBEXECDIR\@|$(libexecdir)|" \
             $< > $@
 
-uidir   = $(datadir)/mate-2.0/ui
+uidir   = $(datadir)/mate-panel/ui
 ui_DATA = brightness-applet-menu.xml
 
 EXTRA_DIST = org.mate.BrightnessApplet.mate-panel-applet.in.in $(ui_DATA) $(service_in_files)
--- mate-power-manager-1.5.1/applets/inhibit/Makefile.am~	2013-03-02 19:33:45.000000000 +0200
+++ mate-power-manager-1.5.1/applets/inhibit/Makefile.am	2013-03-02 19:37:05.210279694 +0200
@@ -59,7 +59,7 @@
             -e "s|\@LIBEXECDIR\@|$(libexecdir)|" \
             $< > $@
 
-uidir   = $(datadir)/mate-2.0/ui
+uidir   = $(datadir)/mate-panel/ui
 ui_DATA = inhibit-applet-menu.xml
 
 EXTRA_DIST = org.mate.InhibitApplet.mate-panel-applet.in.in $(ui_DATA) $(service_in_files)
