--- mate-power-manager-1.5.1/./configure.ac	2013-02-18 13:10:23.719300158 +0200
+++ mate-power-manager-1.5.1/./configure.ac	2013-02-18 13:12:23.171483682 +0200
@@ -106,7 +106,7 @@
 GNOME_KEYRING_REQUIRED=0.6.0
 DBUS_REQUIRED=1.0
 DBUS_GLIB_REQUIRED=0.70
-LIBMATENOTIFY_REQUIRED=1.1.0
+LIBNOTIFY_REQUIRED=0.7.5
 CAIRO_REQUIRED=1.0.0
 UNIQUE_REQUIRED=0.9.4
 LIBPANEL_REQUIRED=1.5.0
@@ -168,7 +168,7 @@
  [xrandr >= 1.3],
  [AC_DEFINE(HAVE_XRANDR_13, 1, [xrandr 1.3 available])])
 
-PKG_CHECK_MODULES(LIBMATENOTIFY, libmatenotify >= $LIBMATENOTIFY_REQUIRED)
+PKG_CHECK_MODULES(LIBNOTIFY, libnotify >= $LIBNOTIFY_REQUIRED)
 
 dnl ----------------------------------------------------------------------------
 dnl - Unique
--- mate-power-manager-1.5.0/src/gpm-manager.c.libnotify	2012-11-08 00:42:58.000000000 +0100
+++ mate-power-manager-1.5.0/src/gpm-manager.c	2013-02-17 03:43:58.448430837 +0100
@@ -40,7 +40,7 @@
 #include <dbus/dbus-glib-lowlevel.h>
 #include <canberra-gtk.h>
 #include <libupower-glib/upower.h>
-#include <libmatenotify/notify.h>
+#include <libnotify/notify.h>
 
 #include "egg-debug.h"
 #include "egg-console-kit.h"
@@ -480,9 +480,9 @@ gpm_manager_notify (GpmManager *manager,
 	/* if the status icon is hidden, don't point at it */
 	if (manager->priv->status_icon != NULL &&
 	    gtk_status_icon_is_embedded (manager->priv->status_icon))
-		notification = notify_notification_new_with_status_icon (title, message, icon, manager->priv->status_icon);
+		notification = notify_notification_new (title, message, gtk_status_icon_get_icon_name(manager->priv->status_icon));
 	else
-		notification = notify_notification_new (title, message, icon, NULL);
+		notification = notify_notification_new (title, message, icon);
 	notify_notification_set_timeout (notification, timeout);
 	notify_notification_set_urgency (notification, urgency);
 	g_signal_connect (notification, "closed", G_CALLBACK (gpm_manager_notification_closed_cb), notification_class);
--- mate-power-manager-1.5.1/./applets/inhibit/Makefile.am~	2012-11-23 04:26:11.000000000 +0200
+++ mate-power-manager-1.5.1/./applets/inhibit/Makefile.am	2013-02-18 13:11:55.123521630 +0200
@@ -4,7 +4,7 @@
 	$(DBUS_CFLAGS)						\
 	$(MATE_CFLAGS)						\
 	$(PANEL_CFLAGS)						\
-	$(LIBMATENOTIFY_CFLAGS)					\
+	$(LIBNOTIFY_CFLAGS)					\
 	$(GTKUNIQUE_CFLAGS)					\
 	-DBINDIR=\"$(bindir)\"			 		\
 	-DMATELOCALEDIR=\""$(datadir)/locale"\"		\
--- mate-power-manager-1.5.1/./applets/brightness/Makefile.am~	2012-11-23 04:26:11.000000000 +0200
+++ mate-power-manager-1.5.1/./applets/brightness/Makefile.am	2013-02-18 13:12:02.690537777 +0200
@@ -4,7 +4,7 @@
 	$(DBUS_CFLAGS)						\
 	$(MATE_CFLAGS)						\
 	$(PANEL_CFLAGS)						\
-	$(LIBMATENOTIFY_CFLAGS)					\
+	$(LIBNOTIFY_CFLAGS)					\
 	$(GTKUNIQUE_CFLAGS)					\
 	-DBINDIR=\"$(bindir)\"			 		\
 	-DMATELOCALEDIR=\""$(datadir)/locale"\"	 	\
--- mate-power-manager-1.5.1/./src/Makefile.am	2013-02-18 13:12:11.624283712 +0200
+++ mate-power-manager-1.5.1/./src/Makefile.am	2013-02-18 13:15:12.969325736 +0200
@@ -15,7 +15,7 @@
 	$(MATE_CFLAGS)					\
 	$(UNIQUE_CFLAGS)				\
 	$(X11_CFLAGS)					\
-	$(LIBMATENOTIFY_CFLAGS)				\
+	$(LIBNOTIFY_CFLAGS)				\
 	$(CANBERRA_CFLAGS)				\
 	$(GSTREAMER_CFLAGS)				\
 	-DI_KNOW_THE_DEVICEKIT_POWER_API_IS_SUBJECT_TO_CHANGE \
@@ -187,7 +187,7 @@
 	$(DBUS_LIBS)					\
 	$(X11_LIBS)						\
 	$(CANBERRA_LIBS)				\
-	$(LIBMATENOTIFY_LIBS)				\
+	$(LIBNOTIFY_LIBS)				\
 	$(GPM_EXTRA_LIBS)				\
 	$(UPOWER_LIBS)					\
 	-lm
@@ -254,7 +254,7 @@
 	$(UPOWER_LIBS)					\
 	$(DBUS_LIBS)					\
 	$(X11_LIBS)						\
-	$(LIBMATENOTIFY_LIBS)				\
+	$(LIBNOTIFY_LIBS)				\
 	$(GPM_EXTRA_LIBS)				\
 	-lm
 
