diff -up mate-power-manager-1.5.0/configure.ac.libnotify mate-power-manager-1.5.0/configure.ac
--- mate-power-manager-1.5.0/configure.ac.libnotify	2013-02-17 03:41:22.511890196 +0100
+++ mate-power-manager-1.5.0/configure.ac	2013-02-17 03:41:22.536889962 +0100
@@ -106,7 +106,7 @@ GDK_REQUIRED=2.11.0
 GNOME_KEYRING_REQUIRED=0.6.0
 DBUS_REQUIRED=1.0
 DBUS_GLIB_REQUIRED=0.70
-LIBMATENOTIFY_REQUIRED=1.1.0
+LIBMATENOTIFY_REQUIRED=0.7.5
 CAIRO_REQUIRED=1.0.0
 UNIQUE_REQUIRED=0.9.4
 LIBPANEL_REQUIRED=1.5.0
@@ -168,7 +168,7 @@ PKG_CHECK_EXISTS(
  [xrandr >= 1.3],
  [AC_DEFINE(HAVE_XRANDR_13, 1, [xrandr 1.3 available])])
 
-PKG_CHECK_MODULES(LIBMATENOTIFY, libmatenotify >= $LIBMATENOTIFY_REQUIRED)
+PKG_CHECK_MODULES(LIBMATENOTIFY, libnotify >= $LIBMATENOTIFY_REQUIRED)
 
 dnl ----------------------------------------------------------------------------
 dnl - Unique
diff -up mate-power-manager-1.5.0/src/gpm-manager.c.libnotify mate-power-manager-1.5.0/src/gpm-manager.c
--- mate-power-manager-1.5.0/src/gpm-manager.c.libnotify	2012-11-08 00:42:58.000000000 +0100
+++ mate-power-manager-1.5.0/src/gpm-manager.c	2013-02-17 03:43:58.448430837 +0100
@@ -40,7 +40,7 @@
 #include <dbus/dbus-glib-lowlevel.h>
 #include <canberra-gtk.h>
 #include <libupower-glib/upower.h>
-#include <libmatenotify/notify.h>
+#include <libnotify/notify.h>
 
 #include "egg-debug.h"
 #include "egg-console-kit.h"
@@ -480,9 +480,9 @@ gpm_manager_notify (GpmManager *manager,
 	/* if the status icon is hidden, don't point at it */
 	if (manager->priv->status_icon != NULL &&
 	    gtk_status_icon_is_embedded (manager->priv->status_icon))
-		notification = notify_notification_new_with_status_icon (title, message, icon, manager->priv->status_icon);
+		notification = notify_notification_new (title, message, gtk_status_icon_get_icon_name(manager->priv->status_icon));
 	else
-		notification = notify_notification_new (title, message, icon, NULL);
+		notification = notify_notification_new (title, message, icon);
 	notify_notification_set_timeout (notification, timeout);
 	notify_notification_set_urgency (notification, urgency);
 	g_signal_connect (notification, "closed", G_CALLBACK (gpm_manager_notification_closed_cb), notification_class);
